base/reducedim.jl

```diff
@@ -17,59 +17,22 @@ reduced_indices(a::AbstractArrayOrBroadcasted, region) = reduced_indices(axes(a)
 # for reductions that keep 0 dims as 0
 reduced_indices0(a::AbstractArray, region) = reduced_indices0(axes(a), region)
 
-function reduced_indices(inds::Indices{N}, d::Int) where N
-    d < 1 && throw(ArgumentError("dimension must be ≥ 1, got $d"))
-    if d == 1
-        return (reduced_index(inds[1]), tail(inds)...)::typeof(inds)
-    elseif 1 < d <= N
-        return tuple(inds[1:d-1]..., oftype(inds[d], reduced_index(inds[d])), inds[d+1:N]...)::typeof(inds)
-    else
-        return inds
-    end
+function reduced_indices(inds::Indices{N}, region_) where N
+    region = _get_valid_region(region_)
+    ntuple(i -> i in region ? reduced_index(inds[i]) : inds[i], Val(N))
```

Comment:
I know Base isn't very consistent, but I'd find this easier to read if it were `axs[d]` not `inds[i]`. (I think `axes(A)` was once upon a time called `indices(A)`.)

Reply:
I would not rename the function. Even if it's technically internal someone may be using it, and there's no gain from gratuitously breaking things.